tiatoolbox/models/controller/semantic_segmentor.py

```diff
@@ -1098,3 +1097,143 @@ def predict(
             self._postproc_workers.shutdown()
         self._postproc_workers = None
         return outputs
+
+
+class FeatureExtractor(SemanticSegmentor):
+    """Generic Feature Extractor."""
```

Comment:
Please can you elaborate further what kind of features this is meant to extract?